xscen/extract.py

```diff
@@ -798,6 +806,15 @@ def search_data_catalogs(
     if restrict_members is not None and len(catalogs) > 0:
         catalogs = _restrict_multimembers(catalogs, id_columns, restrict_members)
 
+    if restrict_warming_level and len(catalogs) > 0:
+        if isinstance(restrict_warming_level, bool):
+            restrict_warming_level = {}
+        if "ignore_member" not in restrict_warming_level:
+            restrict_warming_level["ignore_member"] = False
+        if "tas_csv" not in restrict_warming_level:
+            restrict_warming_level["tas_csv"] = None
```

Comment:
```suggestion
        restrict_warming_level = restrict_warming_level.setdefault("ignore_member", False)
        restrict_warming_level = restrict_warming_level.setdefault("tas_csv", None)
```

Reply:
Careful here. We can do `restrict_warming_level.setdefault("ignore_member", False)` only. If we do `restrict_warming_level =` it will change the dictionnary into a boolean.